h2o-algos/src/main/java/hex/psvm/psvm/MatrixUtils.java

```diff
@@ -95,4 +197,115 @@ public void reduce(ProductMtvTask mrt) {
     }
   }
 
+  /**
+   * Task for Matrix-Vector subtraction.
+   */
+  static class SubtractionMtvTask extends MRTask<SubtractionMtvTask> {
+
+    private Vec v;
+
+    /**
+     * @param v vector size is the same as number of matrix columns
```

Comment:
I think it might be more efficient then to either use double array instead of a Vec or avoid using Vec#at which is expensive (requires a lookup of chunk every time).